Add Regtest support

ysangkok opened the pull request:
But caution, regtest is called TestNet in Lit.
(and testnet is called TestNet3)

For user-facing text, we use the 'regtest' name
to lessen the confusion.

ysangkok commented:
Wait, I just found an issue with the command flag parsing.

kcalvinalvin commented on cmd/util/utils.go:
```diff
@@ -24,39 +24,51 @@ var testNet3GenHash = Hash{
+		case wire.TestNet: // yes, this is regtest
+			return regTestGenHash
+	}
+	panic("net not supported by GenHashForNet")
```
I think returning and error here is better. In some parts of the code, you'll see functions with ```panic(err)``` instead of ```return err``` and this was mostly for debugging since with panics, you get the line # of where it errored out.

I think something like
``` func GenHashForNet(net wire.BitcoinNet) (Hash, error)```
and
``` return nil, fmt.Errorf("net not supported\n")```
is good

kcalvinalvin commented on cmd/util/utils.go:
```diff
@@ -24,39 +24,51 @@ var testNet3GenHash = Hash{
+	f.Read(magicbytes[:])
+
+	bytesToMatch := make([]byte, 4)
+	// Can't use U32tB because we need little-endian
```
In ```cmd/util/utils.go```, there is a function that does little-endian to u32.

I think it would be better if a ```U32tLB()``` was added and called here

kcalvinalvin commented:
A side note but the reason Regtest wasn't really implemented was because Utreexo testing is mostly on the transactions and there aren't really transactions with Regtest, unless of course, if someone set up a fancy Regtest network for themselves.

I think this PR is good because:

1. Simple changes
2. Adds a feature for others
3. Doesn't impede on any other features.

kcalvinalvin commented on cmd/bridgenode/chain.go:
```diff
@@ -4,14 +4,16 @@ import (
 	"os"
 
+	"github.com/mit-dci/lit/wire"
+
```
Please remove the empty line here

ysangkok commented:
@kcalvinalvin all fixed!

kcalvinalvin commented:
@ysangkok I... still see the same code

ysangkok commented:
@kcalvinalvin sorry, for got to "git add -u". fixed now!

kcalvinalvin commented:
Could you also get rid of the empty line in ```cmd/bridgenode/chainio.go``` line number 7

ysangkok commented:
Seems you are against empty lines between groups of dependencies, so I removed them all:

```
diff --git a/cmd/bridgenode/chainio.go b/cmd/bridgenode/chainio.go
index 0f69e34..26daa93 100644
--- a/cmd/bridgenode/chainio.go
+++ b/cmd/bridgenode/chainio.go
@@ -4,7 +4,6 @@ import (
        "os"
 
        "github.com/mit-dci/lit/wire"
-
        "github.com/mit-dci/utreexo/cmd/util"
        "github.com/mit-dci/utreexo/utreexo"
 )
diff --git a/cmd/csn/client.go b/cmd/csn/client.go
index b7d60d4..1ed1d03 100644
--- a/cmd/csn/client.go
+++ b/cmd/csn/client.go
@@ -6,7 +6,6 @@ import (
        "time"
 
        "github.com/mit-dci/lit/wire"
-
        "github.com/mit-dci/utreexo/cmd/util"
        "github.com/syndtr/goleveldb/leveldb"
        "github.com/syndtr/goleveldb/leveldb/opt"
diff --git a/cmd/main.go b/cmd/main.go
index c65781d..e28d0ed 100644
--- a/cmd/main.go
+++ b/cmd/main.go
@@ -8,7 +8,6 @@ import (
        "syscall"
 
        "github.com/mit-dci/lit/wire"
-
        bridge "github.com/mit-dci/utreexo/cmd/bridgenode"
        "github.com/mit-dci/utreexo/cmd/csn"
 )
```

kcalvinalvin commented:
> Seems you are against empty lines between groups of dependencies, so I removed them all:

We should try to adhere to the golang formatting and there should only be an empty line between the standard lib and external imports.

For future reference, if you run ```gofmt "filename"``` it'll output formatting code per golang standards.

ysangkok commented:
ah, right, thanks for the info! i will use a git commit hook. now `gofmt` doesn't report anything, i have indented the switches less.

ysangkok commented:
i have fixed a comment and added a new one:

```
diff --git a/cmd/util/utils.go b/cmd/util/utils.go
index 7e09cfc..2e7d933 100644
--- a/cmd/util/utils.go
+++ b/cmd/util/utils.go
@@ -151,7 +151,7 @@ func BtU32(b []byte) uint32 {
        return i
 }
 
-// uint32 to 4 bytes.  Always works.
+// int32 to 4 bytes (Big Endian).  Always works.
 func I32tB(i int32) []byte {
        var buf bytes.Buffer
        binary.Write(&buf, binary.BigEndian, i)
@@ -171,6 +171,7 @@ func BtI32(b []byte) int32 {
        return i
 }
 
+// uint32 to 4 bytes (Little Endian).  Always works.
 func U32tLB(i uint32) []byte {
        b := make([]byte, 4)
        binary.LittleEndian.PutUint32(b, i)
```

kcalvinalvin commented:
LGTM :)

kcalvinalvin commented:
@adiabat Please review this PR
